### ISSUE TITLE
Multi-Level Navs

### ISSUE THREAD

**Issue by annwa** [rackerlabs/canon#141]
From Dave Williams:
I’m thinking through a major revamp of the cloud office control panel that structurally deviates from the 2 level canon list-detail model in that it has lists, details, and sub-details that are way more complicated than can be jammed into a popover (as canonical doctrine prescribes here). I see that mycloud already deviates with the 3-level server listing / server / monitoring, but I’m not particularly in love with how that works navigationally without a breadcrumb. I’ve also never enjoyed how the ‘back to parent’ link is crammed in there. So how does this seem as an option? Maybe we can discuss on Tues...

![m1-detail-breadcrumb-idea](https://cloud.githubusercontent.com/assets/5985304/5781787/1ce3d742-9d7b-11e4-9537-a9cd134ca35d.png)

**Comment by annwa** [comment]
I think that these navs are much clearer.
MyRack also has the same problem of deep nav.

**Comment by j00k** [comment]
UX did some Customer Experience Workshops a few months ago. I think this issue was something that was noted.

> # The structure forces repetition.
I have to drill down into details and back up to the server list over
and over again. For example, tasks like creating or suppressing
alarms is really tedious because I have to do the same task for every
single server."

**Comment by eddywashere** [comment]
@annwa is it cool to rename to "Updates to Navigation Model" or at least "Updates to Multi Level Nav Model" as it seems to apply directly to that in the [docs](http://rackerlabs.github.io/canon/ux-patterns/#navigation-model) and add a link to that page? We've kind of discussed this problem as "nested list interactions" a few times before but never really came up with a solution outside of the monitoring nav and cloud files nav.

Some of the things we've done in the past include the file tree pattern, simple breadcrumbs and sometimes a nested table. In myrack, specifically managed virt, we expose each nested level as a section in the secondary nav to allow people to interact with similar objects (vms, hypervisors, clusters, etc) across multi levels at the same time. Maybe something like that could help.

@j00k "tasks like creating or suppressing alarms is really tedious because I have to do the same task for every single server." <- seems like less of a nav model problem and more of a component affordance thing / work flow problem / action that can be applied to multiple entities (multi select then actions button).

**Comment by eddywashere** [comment]
I dig the proposed design. Cloud Intelligence has something similar:
![image](https://cloud.githubusercontent.com/assets/399776/5794468/5062b29c-9f23-11e4-87e0-b302238b5efe.png)

**Comment by j00k** [comment]
I agree that workflow in the cp is a different and bigger problem. I just wanted to use that quote because if people are going to be drilling up/down it should be easier.

:+1: breadcrumbs

I like Dave's implementation because it doesn't have a redundant identifier (CI screenshot has the check name in the crumb and header detail).

**Comment by ddubbindustries** [comment]
@eddywashere asked if anything good came from the Open Design Studio discussion about it this week:

I think mostly it was a chance for me to try to describe the context around the problem I'm trying to solve. Since the meeting, I've had an epiphany of how to continue using the 2-level list / detail architecture and I think it is actually a better solution because it brings up the sub-detail items to a more accessible location. It's hard to explain, but at this stage, it looks like the constraint has made for better design. The founding Canonical fathers and mothers knew what they were doing! 

> The more constraints one imposes, the more one frees one's self. And the arbitrariness of the constraint serves only to obtain precision of execution.
> Igor Stravinsky

**Comment by annwa** [comment]
hmm also @ddubbindustries i found this in cloud files.... 
![screen shot 2015-01-30 at 2 22 02 pm](https://cloud.githubusercontent.com/assets/5985304/5982708/820e1b80-a88b-11e4-98f3-a996f33b6260.png)

so it appears if you need a file-nav like structure... there is one. it's just not documented :neutral_face:

**Comment by annwa** [comment]
![screen shot 2015-02-11 at 10 09 42 am](https://cloud.githubusercontent.com/assets/5985304/6150647/225ea142-b1d6-11e4-9917-35412fda138e.png)
to make things more fun my rack also has these....

**Comment by annwa** [comment]
![screen shot 2015-02-11 at 10 10 36 am](https://cloud.githubusercontent.com/assets/5985304/6150655/3efa4d24-b1d6-11e4-9120-42088dc9daaa.png)
